### PR TITLE
Use arrow functions instead of binding anonymous functions

### PR DESCRIPTION
**Pull request by tschaub**
This makes use of arrow functions instead of creating an unnecessary anonymous function and calling bind on it.  It appears that TypeScript does a better job interpreting these arrow functions, so this change required a few updates to types.

### CODE REVIEW

File: src/ol/reproj/DataTile.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ class ReprojDataTile extends DataTile {
 
     /**
      * @private
-     * @type {!Array<import("../Tile.js").default>}
+     * @type {!Array<DataTile>}
```
Comment (tschaub) [inline src/ol/reproj/DataTile.js:119]: The `reproject_` method below assumes these are data tiles, but TypeScript didn't notice with the previous syntax.

File: src/ol/reproj/Tile.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ class ReprojTile extends Tile {
 
     /**
      * @private
-     * @type {!Array<import("../Tile.js").default>}
+     * @type {!Array<ReprojTile|import("../ImageTile.js").default>}
```
Comment (tschaub) [inline src/ol/reproj/Tile.js:106]: Use of `tile.getImage()` below doesn't work with the base tile class.

File: src/ol/webgl/PostProcessingPass.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ const DEFAULT_FRAGMENT_SHADER = `
 /**
  * @typedef {Object} UniformInternalDescription
  * @property {import("./Helper").UniformValue} value Value
- * @property {number} location Location
+ * @property {WebGLUniformLocation} location Location
```
Comment (tschaub) [inline src/ol/webgl/PostProcessingPass.js:49]: TypeScript recognized that this type was wrong after removing bind on anonymous function below.

**Review by ahocevar — APPROVED** [review]

File: src/ol/reproj/Tile.js (hunk 2 above)
Comment (MoonE) [reply src/ol/reproj/Tile.js:106]: Can source tiles be `ReprojTile`s?

I believe the return type of `getTileInternal` should be `TileImage` only.
https://github.com/openlayers/openlayers/blob/66fb11b31679e865711884c19fd48eeeee8b4ca7/src/ol/source/TileImage.js#L356-L365
`createTile_` always returns `TileImage` and the cache should also only contains this type.

Comment (tschaub) [reply src/ol/reproj/Tile.js:106]: @MoonE - Thanks.  I think #14174 addresses your suggestion.

Comment (tschaub) [reply src/ol/reproj/Tile.js:106]: Ooops.  I missed #14173.

**Pull request merged**